Source/Core/Core/NetPlayClient.cpp

```diff
@@ -484,11 +499,13 @@ void NetPlayClient::ThreadFunc()
 	{
 		ENetEvent netEvent;
 		int net;
+		if (m_traversal_client)
+			m_traversal_client->HandleResends();
+		net = enet_host_service(m_client, &netEvent, 4);
+		while (!m_async_queue.Empty())
 		{
-			std::lock_guard<std::recursive_mutex> lks(m_crit.send);
-			if (m_traversal_client)
-				m_traversal_client->HandleResends();
-			net = enet_host_service(m_client, &netEvent, 4);
+			Send(std::move(m_async_queue.Front()));
```

Comment:
I think it's better to not create a new Send(unique_ptr) method, we could just use:
Send(*m_async_queue.Front());
So the memory will be freed on Pop instead of here in Send...